analyses/hello-clusters/01_perform-evaluate-clustering.Rmd

```diff
@@ -49,32 +62,443 @@ data_dir <- file.path(repository_base, "data", "current")
+  pca_matrix,
+  cluster_results_df
+)
+
+# print the result
+stability_results
+```
+
+#### Using non-default clustering parameters
+
+When calculating bootstrap clusters, `rOpenScPCA::calculate_stability()` uses `rOpenScPCA::calculate_clusters()` with default parameters. 
+If your original clusters were not calculated with these defaults, you should pass those customized values into this function as well to ensure a fair comparison between your original clusters and the bootstrap clusters.
+
+
+```{r stability custom parameters}
+# Calculate clusters with non-default parameters
+cluster_df_leiden <- calculate_clusters(
+  pca_matrix,
+  algorithm = "leiden",
+  resolution = 0.5,
+  nn = 15
+)
+
+# Now, pass in the same arguments customizing parameters here
+stability_results_leiden <- calculate_stability(
+  pca_matrix,
+  cluster_df_leiden,
+  algorithm = "leiden",
+  resolution = 0.5,
+  nn = 15
+)
+
+stability_results_leiden
+```
+
+
+
+##  Calculate QC metrics on existing clusters
+
+Even if you perform clustering with other tools, you can still use the `rOpenScPCA` functions to evaluate your clusters.
+
+In this section, we'll present examples of how you can calculate the silhouette width, neighborhood purity, and cluster stability from existing cluster assignments.
+
+### Evaluating Seurat clusters
+
+If you are analyzing your data with a Seurat pipeline that includes calculating clusters, you can use `rOpenScPCA` to evaluate them.
+We previously created a Seurat object called `seurat_obj` with clusters:
+
+
+```{r show seurat clusters}
+head(seurat_obj$seurat_clusters)
+```
+
+To calculate QC metrics on these clusters, we'll need two pieces of information:
+
+- The PCA matrix
+- A data frame with columns `cell_id` and `cluster`
+
+We'll create those here:
+
+```{r prepare seurat input}
+# Extract the PCA matrix from the Seurat object
+seurat_pca_matrix <- Embeddings(seurat_obj, reduction = "pca")
+
+# Create a data frame for input
+seurat_cluster_df <- data.frame(
+  cell_id = names(seurat_obj$seurat_clusters),
+  cluster = seurat_obj$seurat_clusters
+)
+```
+
+Now, we can run `calculate_silhouette()` and `calculate_purity()`:
+
+```{r seurat silhouette}
+seurat_silhouette_df <- calculate_silhouette(
+  seurat_pca_matrix,
+  seurat_cluster_df
+)
+seurat_silhouette_df
+```
+
+```{r seurat purity}
+seurat_purity_df <- calculate_purity(
+  seurat_pca_matrix,
+  seurat_cluster_df
+)
+seurat_purity_df
+```
+
+We do not recommend using `rOpenScPCA::calculate_stability()` on Seurat clusters due to differences in the underlying clustering approach between Seurat and the `bluster` package which `rOpenScPCA` uses.
+
+### Evaluating ScPCA clusters
+
+ScPCA cell metadata already contains a column called `cluster` with results from an automated clustering. 
+These clusters were calculated using `bluster`, the same tool that `rOpenScPCA` uses, with the following parameters:
+
+* The Louvain algorithm
+* Jaccard weighting
+* 20 nearest neighbors
```

Comment:
I would actually show that we have these parameters stored in the object, and use those to retrieve the values:

```
metadata(sce)$cluster_algorithm
metadata(sce)$cluster_weighting
metadata(sce)$cluster_nn
```